Source/OCMock/OCMockObject.m

```diff
@@ -31,15 +31,80 @@
 #import "OCMExpectationRecorder.h"
 #import "OCMQuantifier.h"
 
+@class XCTestCase;
+
+// gCurrentMocksToStopRecorder will point to either gTestCaseMocksToStop or gTestSuiteMocksToStop
+// depending on whether we are currently recoring testcase mocks or test suite mocks.
+// Controlled by OCMockXCTestObserver.
+static NSHashTable<OCMockObject *> *gTestCaseMocksToStop;
+static NSHashTable<OCMockObject *> *gTestSuiteMocksToStop;
+static NSHashTable<OCMockObject *> *gCurrentMocksToStopRecorder;
+
+// Flag that controls whether we should be asserting after stopmocking is called.
+// Controlled by OCMockXCTestObserver.
+static BOOL gAssertOnCallsAfterStopMocking;
+
+// "Fake" Protocol so we can avoid having to link to XCTest, but not get warnings about
+// methods not being declared.
+@protocol OCMockXCTestObservation
++ (id)sharedTestObservationCenter;
+- (void)addTestObserver:(id)observer;
+@end
 
-@implementation OCMockObject
+@interface OCMockXCTestObserver : NSObject
+@end
 
+@implementation OCMockObject
 #pragma mark  Class initialisation
++ (void)load
+{
+    gTestCaseMocksToStop = [[NSHashTable weakObjectsHashTable] retain];
+    gTestSuiteMocksToStop = [[NSHashTable weakObjectsHashTable] retain];
+    gCurrentMocksToStopRecorder = nil;
+    gAssertOnCallsAfterStopMocking = YES;
+    Class xctest = NSClassFromString(@"XCTestObservationCenter");
+    if (xctest)
+    {
+        // If XCTest is available, we set up an observer to stop our mocks for us.
+        [[xctest sharedTestObservationCenter] addTestObserver:[[OCMockXCTestObserver alloc] init]];
+    }
+
+}
 
 + (void)initialize
 {
-	if([[NSInvocation class] instanceMethodSignatureForSelector:@selector(getArgumentAtIndexAsObject:)] == NULL)
-		[NSException raise:NSInternalInconsistencyException format:@"** Expected method not present; the method getArgumentAtIndexAsObject: is not implemented by NSInvocation. If you see this exception it is likely that you are using the static library version of OCMock and your project is not configured correctly to load categories from static libraries. Did you forget to add the -ObjC linker flag?"];
+    if (self == [OCMockObject class]) {
```

Comment:
Is this just an unrelated clean-up? I'm asking because I know about `initialize` semantics and thought about adding that `if` statement earlier myself. I decided against it though, because in the end it doesn't add any value I could think of. If the method isn't there you get the exception. If it's there you're potentially testing it more than once, right?

Reply:
Originally the code in the +load was located here and I didn't want it getting initialized more than once. I usually invert your idea and always put an `if` in my  +initialize because I'm afraid that other developers may not understand the initialize semantics and assume it will be called only once. I'll remove as it adds nothing to this particular CL.